packages/devtools_app/lib/src/debugger/flutter/console.dart

```diff
@@ -30,7 +33,102 @@ class Console extends StatefulWidget {
 }
 
 class _ConsoleState extends State<Console> {
-  final scrollController = ScrollController();
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<List<String>>(
+      valueListenable: widget.controller.stdio,
+      builder: (context, lines, _) {
+        // Extract this to a reusable ConsoleWithControls StatelessWidget?
+        return Stack(
+          children: [
+            ConsoleOutput(lines: lines),
+            Container(
+              alignment: Alignment.bottomRight,
+              child: ConsoleControls(
+                disabled: lines.isEmpty,
+                onCopyPressed: () {
+                  Clipboard.setData(ClipboardData(
+                    text: lines.join('\n'),
+                  )).then((_) {
+                    Notifications.of(context)?.push(
+                      'Copied!',
+                    );
+                  });
+                },
+                onDeletePressed: widget.controller.clearStdio,
+              ),
+            ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+/// Optionally renders a ButtonBar with Console Controls,
+/// Copy and Clear.
+/// The callbacks for those buttons are passed from the outside.
+class ConsoleControls extends StatelessWidget {
+  const ConsoleControls({
+    this.disabled,
+    this.onCopyPressed,
+    this.onDeletePressed,
+  }) : super();
+
+  final Function onCopyPressed;
+  final Function onDeletePressed;
+  final bool disabled;
+
+  @override
+  Widget build(BuildContext context) {
+    return disabled
+        ? Container()
+        : ButtonBar(
+            buttonPadding: EdgeInsets.zero,
+            alignment: MainAxisAlignment.end,
+            children: [
+              IconButton(
+                icon: const Icon(Icons.content_copy),
+                onPressed: onCopyPressed,
+                tooltip: 'Copy to clipboard.',
+              ),
+              IconButton(
+                icon: const Icon(Icons.block),
```

Comment:
Use `Icons.delete`? This is what we use to remove all breakpoints in the breakpoints view.

Reply:
I used `delete`, but then I saw the "Clear" in the "Logging" panel using `block`, and changed it last minute (after I took the screenshot, btw! :P)

Reply:
Gotcha (this is closer to the logging console than the breakpoints view)

Reply:
Changed to `delete`